pcb.c

```diff
@@ -0,0 +1,473 @@
+/**
+ * Date: 5/4/25
+ * File: pcb.c
+ * Description: PCB implementation for Yalnix OS
+ */
+
+#include "pcb.h"
+
+// Global variables
+list_t *ready_queue;
+list_t *delay_queue;
+list_t *blocked_queue;
+list_t *zombie_queue;
+pcb_t *current_process = NULL;
+int next_pid = 1;  // Start PIDs at 1
+
+int init_pcb_system(void) {
+    TracePrintf(1, "ENTER init_pcb_system.\n");
+    ready_queue = create_list();
+    delay_queue = create_list();
+    blocked_queue = create_list();
+    zombie_queue = create_list();
+
+    // If any of the queues failed to initialize return Error, else return 0
+    if(ready_queue == NULL || delay_queue  == NULL || zombie_queue  == NULL || blocked_queue == NULL) {
+        TracePrintf(1, "ERROR, The kernel has failed to allocate a pcb queue.\n");
+        return ERROR;
+    }
+    TracePrintf(1, "EXIT init_pcb_system.\n");
+    return 0;
+}
+
+pcb_t *create_pcb(void) {
+    TracePrintf(1, "ENTER create_pcb.\n");
+    // Allocate memory for new PCB
+    pcb_t *new_pcb = malloc(sizeof(pcb_t));
+    if (new_pcb == NULL) {
+        TracePrintf(1, "ERROR, The kernel has failed to allocate the pcb.\n");
+        return NULL;
+    } // If it failed return NULL
+
+    // Assign unique PID
+    new_pcb->pid = next_pid++;

```

Comment:
you should use the helper_new_pid function so that you can associate the pid with a page table, to assist with debugging.

Reply:
you could take the page table as an argument if you wanted